Add documentation for "Getting Started" chapter (refs EVEREST-950) (#6)
chargebyte/chargebyte-docs: docs/source/getting_started.rst

with the Charge Control C and the EVerest charging stack. For this purpose, a basic AC PWM charger
is set up as an example and explained step by step.

Setting Up the Hardware
------------------------

Before you start setting up the hardware, please read the :ref:`safety_notes`.

> 

Reviewer: one more empty line

or via serial interface. The following sections explain how to connect to the Charge Control C using
PuTTY. You can also use other terminal programs (like e.g. MobaXTerm) to connect to the
Charge Control C.

SSH Connection with PuTTY
^^^^^^^^^^^^^^^^^^^^^^^^^

Here are the steps to connect to the Charge Control C via SSH using PuTTY:

#. Install PuTTY on your computer. You can download PuTTY from the following link:
   `PuTTY Download <https://www.chiark.greenend.org.uk/~sgtatham/putty/latest.html>`_.
#. Connect the Charge Control C over Ethernet to your computer.
#. Start PuTTY and enter the IPv4 fallback address "169.254.12.53" of the Charge Control C in
   the "Host Name (or IP address)" field (See figure :ref:`PuTTY SSH Configuration <PuTTY_Serial_Configuration_2>`).
#. Click on "Save" to save the configuration and then click on "Open" to start the connection.
#. A PuTTY Security Alert window will appear. Click on "Accept" to continue.
#. Enter the username "root" and the password "zebematado" to log in to the Charge Control C.

.. note::
   The username can also be stored under "Connection -> Data -> Auto-login username" in the
   PuTTY configuration.

.. note::

>    It is also possible to install a SSH key on the Charge Control C to log in without a

Reviewer 2: an SSH key
```suggestion
   It is also possible to install an SSH key on the Charge Control C to log in without a
```

   2024-06-19T19:26:15.322579+0200 tarragon manager[11999]: [INFO] tarragon_bsp:Cb  :: handle_enable: Setting new duty cycle of 100.00%
   2024-06-19T19:26:15.333875+0200 tarragon manager[11999]: [INFO] tarragon_bsp:Cb  :: CP state change from PowerOn to A, U_CP+: 11947 mV, U_CP-: 7 mV
   2024-06-19T19:26:15.459775+0200 tarragon manager[11999]: [INFO] tarragon_bsp:Cb  :: handle_pwm_off: Setting new duty cycle of 100.00%
   2024-06-19T19:26:15.632804+0200 tarragon manager[11999]: [INFO] tarragon_bsp:Cb  :: handle_pwm_off: Setting new duty cycle of 100.00%

Before plugging in the IEC 62196 Type 2 EVSE Test Adapter, please make sure that the CP state of the
EVSE Test Adapter is set to "B" and the current limit (adjustable via the PP state) is configured to
16A. Now you are ready to start the first charging session with EVerest and you can plug in the IEC
62196 Type 2 EVSE Test Adapter.

After plugging in the IEC 62196 Type 2 EVSE Test Adapter, a CP state change from "A" to "B" should
be visible in the EVerest log. By default, the authentication is disabled in the bsp-only.yaml
configuration.

> Therefore, the duty dycle should directly switch from 100% to ~26.7%. The duty cycle change indicates

Reviewer: cycle

Setting Up the Hardware
------------------------

> Before you start setting up the hardware, please read the safty instructions in the

Reviewer: safety

- Charge Control C (100, 200 or 300)
- 12 V DC Power Supply
- Contactor

> - Power Contactor

Reviewer: What is the different between Contactor and Power Contactor in this context?
I would write Contactor with feedback contact.

Author: Just a misstake while merging from everest/charge_control_c

- IEC 62196 Type 2 three-phase EV charging socket outlet
- Ethernet cable for SSH connection or USB to serial adapter for serial connection
- IEC 62196 Type 2 EVSE Test Adapter (e.g. Metrel or Benning) to simulate the EV
- Wiring material

Hardware Overview
^^^^^^^^^^^^^^^^^

The following figure shows the basic setup of the AC PWM charger with the Charge Control C:

> .. image:: _static/images/ac_pwm_charger_ccc_setup.svg

Author: what exactly does not fit with the drawing

> .. image:: _static/images/ac_pwm_charger_ccc_setup.svg

Reviewer 3: Sorry, i forgot to mention. The drawing uses the contactor feedback while the configuration has no contactor feedback.

    :width: 500pt
    :align: center

.. raw:: html

   <div style="text-align: center;">
      Figure: Basic Setup of the AC PWM Charger with the Charge Control C

>    </div>

Reviewer: You can use .. figure directive with caption feature to get rid of these lines. I added a centered text alignment in custom css/cb_theme.css file. This will place figure's caption under the pic.

Note: Before you start setting up the setup, please check whether the HW components used are also
listed in `Hardware Components section`_.

.. _Hardware Components section: #hardware-components

First Startup
-------------

Boot Process
^^^^^^^^^^^^

Here some key points about the boot process of the Charge Control C:

> - The file system basically consists of three ext4 partionions. Two partitions are used as slots for

Reviewer: partionions -> partitions

>   RAUC update process. The third partion is not not touched by the RAUC update process and is usualy

Reviewer: partion -> partition

>   used to the store update bundles, logs, etc. For more information about the RAUC update process

Reviewer: to store

Reviewer 2: of "for storing"

  see the TODO: ADD LINK HERE section.
- After connecting the Charge Control C to the power supply, the U-Boot bootloader starts the
  currently active slot managed by RAUC.
- The LED status indicators on the Charge Control C provide information about the current status of
  the boot process.
- EVerest is automatically started with the default configuration of chargebyte after the boot

>   process is completed. The inital configuration is explained in the `Initial Configuration`_ section.

Reviewer 2: inital -> initial
```suggestion
  process is completed. The initial configuration is explained in the `Initial Configuration`_ section.
```

|                          +---------------------------------+------------------------------------+
|                          | LED2 (yellow)                   | off                                |
|                          +---------------------------------+------------------------------------+
|                          | LED3 (red)                      | blinking                           |
+--------------------------+---------------------------------+------------------------------------+

Connecting via SSH or Serial Interface
--------------------------------------

For debugging there are two ways to connect the developer computer to the Charge Control C: via SSH
or via serial interface. The following sections explain how to connect to the Charge Control C using
PuTTY. You can also use other terminal programs (like e.g. MobaXTerm) to connect to the
Charge Control C.

SSH Connection with PuTTY
^^^^^^^^^^^^^^^^^^^^^^^^^

Here are the steps to connect to the Charge Control C via SSH using PuTTY:

#. Install PuTTY on your computer. You can download PuTTY from the following link:
   `PuTTY Download <https://www.chiark.greenend.org.uk/~sgtatham/putty/latest.html>`_.

> #. Connected the Charge Control C over Ethernet to your computer

Reviewer: Connect

#. Start PuTTY and enter the IPv4 fallback address "169.254.12.53" of the Charge Control C in

>    the "Host Name (or IP address)" field (See figure :ref:`PuTTY SSH Configration <PuTTY_Serial_Configuration_2>`)..

Reviewer 2: Configration -> Configuration
```suggestion
   the "Host Name (or IP address)" field (See figure :ref:`PuTTY SSH Configuration <PuTTY_Serial_Configuration_2>`)..
```

> #. Click on "Save" to save the configuration ans then click on "Open" to start the connection.

Reviewer 2: ans -> and
```suggestion
#. Click on "Save" to save the configuration and then click on "Open" to start the connection.
```

> #. An PuTTY Security Alert window will appear. Click on "Accept" to continue.

Reviewer 2: A PuTTY
```suggestion
#. A PuTTY Security Alert window will appear. Click on "Accept" to continue.
```
Maybe mention which security alert (host key, I guess), and why it's okay to accept.

#. Enter the username "root" and the password "zebematado" to log in to the Charge Control C.

Note: The user name can also be stored under "Connection -> Data -> Auto-login username" in the
PuTTY configuration.

> Note: It is also possible to install a SSH key on the Charge Control C to log in without a

Reviewer 2: -> an SSH key
```suggestion
Note: It is also possible to install an SSH key on the Charge Control C to log in without a
```

> password. For more information, see the TODO: Link section.

Reviewer 2: TODO 😉

.. image:: _static/images/putty_config_ssh_fallback_ip.png
    :width: 300pt
    :align: center
    :name: PuTTY_SSH_Configuration

.. raw:: html
 
   <div style="text-align: center;">

>      Figure: PuTTY SSH Configration

Reviewer 2: Configration -> Configuration
```suggestion
     Figure: PuTTY SSH Configuration
```

#. Switch back to the "Session" and click on "Save" to save the configuration
   (See figure :ref:`PuTTY Save Serial Connection <PuTTY_Serial_Configuration_2>`).
#. Click on "Open" to start the connection.
#. Now a black window will appear. Press enter to get the login prompt.
#. Enter the username "root" and the password "zebematado" to log in to the Charge Control C.

.. image:: _static/images/putty_config_serial_1.png
    :width: 300pt
    :align: center
    :name: PuTTY_Serial_Configuration_1

.. raw:: html

   <div style="text-align: center;">
     Figure: PuTTY Serial Configuration
   </div>

.. image:: _static/images/putty_config_serial_2.png
    :width: 300pt
    :align: center
    :name: PuTTY_Serial_Configuration_2

.. raw:: html

   <div style="text-align: center;">
     Figure: PuTTY Save Serial Connection
   </div>

Initial Configuration
---------------------

Now you are connected to the Charge Control C and we can have a deeper look at the initial
configuration.

The conifguration files of the EVerest charging stack are stored in the directory "/etc/everest".
EVerest uses the YAML format for the configuration files. The default configuration file of EVerest

> is the config.yaml. If you take a look at the content of the configuration file, you will see that

Reviewer 2: no "the".
```suggestion
is "config.yaml". If you take a look at the content of the configuration file, you will see that
```

Reviewer: ping

there is only a reference to the "bsp-only.yaml" file.

Note: In case you create an own configuration file, you can also store it in the "/etc/everest"

> directory and create a simbolic link to it like

Reviewer: symbolic

"ln -sf /etc/everest/my-config.yaml /etc/everest/config.yaml".

.. code-block:: bash

   root@chargecontrolc:~# ls -l /etc/everest
   total 8
   -rw-r--r-- 1 root root  0 Jan  1  1970 bsp-only.yaml

>    lrwxrwxrwx 1 root root 12 Jan  1  1970 my-config.yaml  -> config.yaml

Reviewer 2: Why does bsp-only.yaml have a zero size here?

And, if I understand correctly, config.yaml is usually a symbolic link to another file, not vice versa.

Reviewer: On my Tarragon bsp-only.yaml has size 1093

Reviewer 2: This still looks unrealistic.

Let's take a look at the content of the bsp-only.yaml configuration file. This file is already
prepared for the basic AC PWM charger setup.

Just type "less /etc/everest/bsp-only.conf" to see the content of the configuration file:

.. literalinclude:: _static/files/bsp-only.yaml
   :language: yaml
   :linenos:

> In general, the EVerest charging stack consists of different modules and each of which fulfils a

Reviewer: fulfills

> In general, the EVerest charging stack consists of different modules and each of which fulfils a

Reviewer 2: ```suggestion
In general, the EVerest charging stack consists of different modules, each of which fulfils a
```

specific task. An EVerest module provides and requests interfaces. The configuration file shows
which EVerest modules are activated, how they are configured and how they are connected to each
other over the interfaces.

However, not all configuration parameters of the modules are shown here. Only the configuration
parameters that do not match the default configuration of the respective module need
to be specified here. Depending on the installed hardware components, the configuration file may

> need to be adapted. The hardware related tasks are maninly handled by the CbTarragonDriver module.

Reviewer: mainly

Reviewer: typo not solved

> need to be adapted. The hardware related tasks are maninly handled by the CbTarragonDriver module.

Reviewer 2: On Charge Control C, most hardware related tasks are handled...

> The configuration of the CbTarragonDriver module can be found in "/usr/libexec/everest/modules/cbTarragonDriver"

Reviewer 2: This is a bit confusing, because the actual configuration is in /etc/everest/config.yaml. The manifest reflects the accepted configuration values and defines the defaults (or does it? are the defaults determined at compile time? not sure).

directory.

> Each module has a specific configuration file, this file is called "mainifest.yaml" and is stored 

Reviewer: manifest

in the main directory of the module.
Here you can also see all other configuration parameters of the respective module. 
Now please type "less /usr/libexec/everest/modules/cbTarragonDriver/manifest.yaml" to see the
content of the configuration file and check whether the configuration fits to your hardware setup.

If you want to change a configuration parameter of a module, which is not part of your EVerest yaml
configuration file, just copy the specific configuration key from the "manifest.yaml" file of the
module to the module specific "config_module" space in your EVerest configuration and adjust the

> value. Please note if you change it directly in the "manifest.yaml" file of a mondule, the changes

Reviewer: module

will be get lost after a software update.

> Here is excerpt of an EVerst configuration to change the parameter "connector_type" to

Reviewer: EVerest

"IEC62196Type2Cable" of the CbTarragonDriver module.

.. code-block:: sh

  tarragon_bsp:
    module: CbTarragonDriver
    config_module:
      contactor_1_feedback_type: none
      relay_2_name: none
      connector_type: IEC62196Type2Cable

After adjusting the configuration file, you have to restart the EVerest charging stack to apply the
changes. Just type "systemctl restart everest" to restart the EVerest charging stack.

Note: You can also use the `EVerest admin panel <https://github.com/EVerest/everest-admin-panel>`_

> to adjust the EVerest configurationin a GUI. This tool must be currently installed manually on your

Reviewer: configuration in

> to adjust the EVerest configurationin a GUI. This tool must be currently installed manually on your

Reviewer 2: Word order:
```suggestion
to adjust the EVerest configurationin a GUI. This tool must currently be installed manually on your
```

> developer computer, because the recources on the board are limited.

Reviewer: resources

Note: If you have made a mistake in the configuration file, the EVerest charging stack will not
start. Therefore, it is recommended to back up the original configuration file before making
changes.

Starting and Monitoring the Charging Process
--------------------------------------------

> Before we are starting the first charging session, it is possible to monitor the charging process

Reviewer 2: Before we start
```suggestion
Before we start the first charging session, it is possible to monitor the charging process
```

Also, a bit confusing: How can we monitor the charging process, before charging session has even started? 😉

   2024-06-19T19:26:14.889409+0200 tarragon manager[11999]: [INFO] tarragon_bsp:Cb  :: Proximity Pilot Observation Thread started
   2024-06-19T19:26:14.957703+0200 tarragon manager[11996]: [INFO] connector:EvseM  :: Max AC hardware capabilities: 32A/3ph
   2024-06-19T19:26:15.164664+0200 tarragon manager[11996]: [INFO] connector:EvseM  :: 🌀🌀🌀 Ready to start charging 🌀🌀🌀
   2024-06-19T19:26:15.322579+0200 tarragon manager[11999]: [INFO] tarragon_bsp:Cb  :: handle_enable: Setting new duty cycle of 100.00%
   2024-06-19T19:26:15.333875+0200 tarragon manager[11999]: [INFO] tarragon_bsp:Cb  :: CP state change from PowerOn to A, U_CP+: 11947 mV, U_CP-: 7 mV
   2024-06-19T19:26:15.459775+0200 tarragon manager[11999]: [INFO] tarragon_bsp:Cb  :: handle_pwm_off: Setting new duty cycle of 100.00%
   2024-06-19T19:26:15.632804+0200 tarragon manager[11999]: [INFO] tarragon_bsp:Cb  :: handle_pwm_off: Setting new duty cycle of 100.00%

Before plugging in the IEC 62196 Type 2 EVSE Test Adapter, please make sure that the CP state of the
EVSE Test Adapter is set to "B" and the current limit (adjustable via the PP state) is configured to

> 16A. Now you are ready to start the first charging session with EVerest and you can plugin the IEC

Reviewer 2: -> plug in
```suggestion
16A. Now you are ready to start the first charging session with EVerest and you can plug in the IEC
```

62196 Type 2 EVSE Test Adapter.

After plugging in the IEC 62196 Type 2 EVSE Test Adapter, a CP state change from "A" to "B" should

> be visible in the EVerest log. Per default the authentication is disabled in the bsp-only.yaml.

Reviewer 2: ```suggestion
be visible in the EVerest log. Per default the authentication is disabled in the bsp-only.yaml configuration.
```

.. code-block:: sh

   2024-06-21T07:45:50.605759+0200 tarragon manager[5733]: [INFO] tarragon_bsp:Cb  :: CP state change from B to C, U_CP+: 5996 mV, U_CP-: -11817 mV
   2024-06-21T07:45:50.766409+0200 tarragon manager[5733]: [INFO] tarragon_bsp:Cb  :: Closing contactor...
   2024-06-21T07:45:50.766409+0200 tarragon manager[5733]: [INFO] tarragon_bsp:Cb  :: Contactor state: CLOSED

The charging process can be stopped by a CP state change from "C" to "B" via the IEC 62196 Type 2.

The last EVerest log messages should look like this:

.. code-block:: sh

   2024-06-21T07:48:04.406373+0200 tarragon manager[5733]: [INFO] tarragon_bsp:Cb  :: CP state change from C to B, U_CP+: 8842 mV, U_CP-: -11826 mV
   2024-06-21T07:48:04.641265+0200 tarragon manager[5733]: [INFO] tarragon_bsp:Cb  :: Opening contactor...
   2024-06-21T07:48:04.641265+0200 tarragon manager[5733]: [INFO] tarragon_bsp:Cb  :: Contactor state: OPEN

Now the charging process is stopped and the IEC 62196 Type 2 EVSE Test Adapter can be unplugged from
the socket outlet or the charging process can be started again by a CP state change from "B" to "C".

> The last EVerest log messages after removing of the plug should look like this:

Reviewer 2: removing of the plug ->
removing the plug, or
removal of the plug

.. code-block:: sh

   2024-06-21T07:52:51.287007+0200 tarragon manager[5733]: [INFO] tarragon_bsp:Cb  :: CP state change from B to A, U_CP+: 11927 mV, U_CP-: -11970 mV
   2024-06-21T07:52:51.368428+0200 tarragon manager[5733]: [INFO] tarragon_bsp:Cb  :: handle_pwm_off: Setting new duty cycle of 100.00%

> Congratulations! You have successfully started your first charging session with the EVerest

Reviewer 2: Not only started, but also finished. So I might use the word "completed".
```suggestion
Congratulations! You have successfully completed your first charging session with the EVerest
```

charging stack and the Charge Control C. Now you are prepared to start your own charging project and

> andjust your setup to your needs. The following chapters will help you to understand the EVerest

Reviewer: adjust

    Figure: Basic Setup of the AC PWM Charger with the Charge Control C

> .. note:

Reviewer: .. note::

#. Install PuTTY on your computer. You can download PuTTY from the following link:
   `PuTTY Download <https://www.chiark.greenend.org.uk/~sgtatham/putty/latest.html>`_.
#. Connect the Charge Control C over Ethernet to your computer.
#. Start PuTTY and enter the IPv4 fallback address "169.254.12.53" of the Charge Control C in
   the "Host Name (or IP address)" field (See figure :ref:`PuTTY SSH Configuration <PuTTY_Serial_Configuration_2>`).
#. Click on "Save" to save the configuration and then click on "Open" to start the connection.
#. A PuTTY Security Alert window will appear. Click on "Accept" to continue.
#. Enter the username "root" and the password "zebematado" to log in to the Charge Control C.

Note: The username can also be stored under "Connection -> Data -> Auto-login username" in the
PuTTY configuration.

Note: It is also possible to install a SSH key on the Charge Control C to log in without a
password. Please look online for instructions on how to do this on a Linux system.

.. figure:: _static/images/putty_config_ssh_fallback_ip.png
    :width: 300pt

>     Figure: PuTTY SSH Configration

Reviewer: Configuration

#. A PuTTY Security Alert window will appear. Click on "Accept" to continue.
#. Enter the username "root" and the password "zebematado" to log in to the Charge Control C.

> .. note:

Reviewer: dito

   The username can also be stored under "Connection -> Data -> Auto-login username" in the
   PuTTY configuration.

> .. note:

Reviewer: dito ....

   2024-06-21T07:48:04.641265+0200 tarragon manager[5733]: [INFO] tarragon_bsp:Cb  :: Contactor state: OPEN

Now the charging process is stopped and the IEC 62196 Type 2 EVSE Test Adapter can be unplugged from
the socket outlet. Alternatively, the charging process can be started again by a CP state change
from "B" to "C".

The last EVerest log messages after removing the plug should look like this:

.. code-block:: sh

   2024-06-21T07:52:51.287007+0200 tarragon manager[5733]: [INFO] tarragon_bsp:Cb  :: CP state change from B to A, U_CP+: 11927 mV, U_CP-: -11970 mV
   2024-06-21T07:52:51.368428+0200 tarragon manager[5733]: [INFO] tarragon_bsp:Cb  :: handle_pwm_off: Setting new duty cycle of 100.00%

Congratulations! You have successfully completed your first charging session with the EVerest
charging stack and the Charge Control C. Now you are prepared to start your own charging project and

> and adjust your setup to your needs. The following chapters will help you understand the EVerest

Reviewer: two times "and"
